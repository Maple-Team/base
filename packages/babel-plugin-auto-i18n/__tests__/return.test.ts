import { getTransformCode } from './test-help'

describe('ReturnStatement', () => {
  it('', () => {
    const sourceCode = `
   export const Component = () => {
    const errorMessage = useMemo(() => {
      if (!errorCode) return ''
      switch (errorCode) {
        case MediaError.MEDIA_ERR_NETWORK:
          return '当前视频加载网络错误'
        case MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED:
          return '当前视频地址错误或格式不支持'
        default:
          return '当前视频出现其他错误'
      }
    }, [errorCode])
    return <div>{errorMessage}</div>
   }
    `
    const result = getTransformCode(sourceCode, 'filename.tsx', false)
    expect(result?.code).toMatchSnapshot()
  })
})
