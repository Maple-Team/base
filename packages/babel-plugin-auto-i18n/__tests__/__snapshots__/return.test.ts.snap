// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnStatement  1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var errorMessage = useMemo(function () {
    if (!errorCode) return '';
    switch (errorCode) {
      case MediaError.MEDIA_ERR_NETWORK:
        return t(\\"key_当前视频加载网络错误\\");
      case MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED:
        return t(\\"key_当前视频地址错误或格式不支持\\");
      default:
        return t(\\"key_当前视频出现其他错误\\");
    }
  }, [errorCode, t]);
  return <div>{errorMessage}</div>;
};"
`;
